app/src/main/java/com/sdp/swiftwallet/presentation/main/fragments/PaymentFragment.java

```diff
@@ -4,17 +4,41 @@
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Spinner;
+import android.widget.TextView;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
 
 import com.sdp.cryptowalletapp.R;
+import com.sdp.swiftwallet.SwiftWalletApp;
+import com.sdp.swiftwallet.data.repository.Web3Requests;
+import com.sdp.swiftwallet.domain.model.QRCodeScanner;
 
 /**
- * A simple {@link Fragment} subclass.
- * Use the {@link PaymentFragment#} factory method to
- * create an instance of this fragment.
+ * Payment fragment to send some cryptocurrency assets
  */
 public class PaymentFragment extends Fragment {
+    private ArrayAdapter<String> arrayAdapter;
+    private TextView fromAddress;
+    private TextView fromBalance;
+    private TextView toAddress;
+
+    private final OnFromAddressSelected onFromAddressSelected = new OnFromAddressSelected();
+    QRCodeScanner qrCodeScanner = new QRCodeScanner(this::setToSelectedAddress, this);
+    private Web3Requests web3Requests;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        SwiftWalletApp application = (SwiftWalletApp) requireActivity().getApplication();
```

Comment:
If tests for this Activity does need Hilt, it will crash almost surely i think

Reply:
Why would it crash ?

Reply:
because of the cast (SwiftWalletApp)